src/libcrun/utils.c

```diff
@@ -2146,3 +2146,15 @@ base64_decode (const char *iptr, size_t isize, char *optr, size_t osize, size_t
     }
   return consumed;
 }
+
+char *
+get_user_name (uid_t uid)
+{
+  struct passwd *pw = getpwuid (uid);
```

Comment:
`getpwuid` is not thread safe.  Let' use `getpwuid_r`.

Also I think we need to `xstrdup()` the result since the caller is going to free it after usage

Reply:
@giuseppe Applied relevant changes, could you please review again.